mongoengine/fields.py

```diff
@@ -619,6 +619,14 @@ class DynamicField(BaseField):
 
     Used by :class:`~mongoengine.DynamicDocument` to handle dynamic data"""
 
+    def __init__(self, container_class=dict, *args, **kwargs):
+        self._container_cls = container_class
+        if not issubclass(self._container_cls, dict):
```

Comment:
We need to check against any Mapping instead. If it behaves like a dictionary it's good enough for me.